Lib/pdb.py

```diff
@@ -1536,19 +1538,50 @@ def complete_unalias(self, text, line, begidx, endidx):
 
     def print_stack_trace(self):
         try:
-            for frame_lineno in self.stack:
-                self.print_stack_entry(frame_lineno)
+            for frame_idx in range(len(self.stack)):
+                self.print_stack_entry(frame_idx)
         except KeyboardInterrupt:
             pass
 
-    def print_stack_entry(self, frame_lineno, prompt_prefix=line_prefix):
-        frame, lineno = frame_lineno
+    def print_stack_entry(self, frame_idx, prompt_prefix=line_prefix):
```

Comment:
This is a breaking change in terms of method signature. That would mean we would only be able to accept this change for 3.12. If this is what you want, you can keep the change, as it makes using the method safer.